doc/Lifecycling.md

```diff
@@ -0,0 +1,223 @@
+
+```Haskell
+When (t ≥ t_1) (Scale (Observe "S" - K) (When (t ≥ t_2) (One “USD”)))
+```
+
+which we acquired at a time `t_0` with `t_0` ≤ `t_1` ≤ `t_2`.
+
+Economically, this contract observes a certain value `S(t)` at time `t_1` and pays the contract holder an amount `S(t_1) - K` in dollars at time `t_2`. Many will recognize in this expression the payoff of an equity forward contract, where `S` is the stock's spot price and `K` is the strike price.
+
+In order to understand how the contract evolves over time, we need to recall that `When pred c` acquires the underlying contract on the first instant the predicate `pred` becomes `True`.
+
+Thus, exactly at time `t_1`, the boolean condition on the outer `When` is met and we acquire `c2`
+
+```Haskell
+Scale (Observe "S" - K) (When (t ≥ t_2) (One “USD”))
+```
+
+as a replacement for the initial contract.
+
+As soon as we acquire `c2`, because of the `Scale` node, we acquire `c3`
+
+```Haskell
+(S(t_1) - K) * (When (t ≥ t_2) (One “USD”)
+```
+
+as a replacement for `c2`.
+
+Finally, once `t_2` has arrived, the predicate on the remaining `When` is met and we acquire `c4`
+
+```Haskell
+(S(t_1) - K) * (One “USD”)
+```
+
+as a replacement for `c3`.
+
+This contract does not include additional dynamics and will not evolve further. In a financial setting, this is when a payment is made to the contract owner for the amount due:
+
+- S(t_1) - K USD is paid to the contract holder
+
+- `c4` is replaced with `Zero`, the contract indicating the absence of a claim.
+
+## Lifecycle functionality
+
+In order to replicate this evolution process for a generic contract, we need to address two problems:
+
+- given a contract, we need a way to figure out when a relevant event has happened (specifically, the next relevant event date)
+
+- then, we need to be able to evolve the contract according to the rules defined by our contract primitives
+
+The lifecycling functionality of the library addresses the latter problem.
+
+### Transient vs Stable nodes
+
+As we can see from the example above, there are some nodes that, once acquired, are immediately consumed and trigger the acquisition of another contract.
+
+A transient contract is a contract whose outer node is transient.
+
+Transient nodes:
+
+- `Scale`, `One`, `Or`, `And`, `Cond`, `Give`
+- `When pred c` when `pred` is `True` at the time of acquisition
+- `Until pred c` when `pred` is `True` at the time of acquisition
+- `Until pred c` when `c` is transient contract
+
+Stable nodes:
+
+- `Zero`, `Anytime`
+- `When pred c` when `pred` is `False` at the time of acquisition
+- `Until pred c` when `pred` is `False` at the time of acquisition and `c` is not transient
+
+As a rule of thumb, the lifecycling function shall not return a transient contract.
+
+### Types of event dates
+
+Assuming that, once we acquire a `One a`, this is immediately paid to the contract holder, we can classify event dates (or time instants) as
```

Comment:
You have this nice taxonomy of dates here, but you introduce acquistion dates above.

Reply:
Effectively the acquisition time is introduced in the PJ paper